### PR TITLE
fix(dashboard): Fix multiple queries on dashboard

### PR DESCRIPTION
**Pull request by billyvg**
There were a few race conditions when dashboards would get re-rendered
(e.g. changing date). Parent component would fetch releases, as well as
global state change, and a withRouter render as well. Also query builders
were not being re-built. Also adds an additional integration test.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx [hunk 1]
```diff
@@ -68,25 +87,45 @@ class DiscoverQuery extends React.Component {
...
+  );
+
   createQueryBuilders() {
     const {organization, queries} = this.props;
+
+    this.queryBuilders = [];
+
     queries.forEach(({constraints, ...query}) => {
       if (constraints && constraints.includes('recentReleases')) {
+        // Can't create query yet because no releases
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx]: This feels like a class of `busy` status situations - should the parent be passing busy=true rather that relying on this component to check here whether there are release constraints and the releases are not loaded?

File: src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx [hunk 2]
```diff
@@ -9,16 +9,27 @@ import {parsePeriodToHours} from 'app/utils';
 import SentryTypes from 'app/sentryTypes';
 import createQueryBuilder from 'app/views/organizationDiscover/queryBuilder';
 
+// Note: Limit max releases so that chart is still a bit readable
+const MAX_RECENT_RELEASES = 20;
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx]: 👍

File: src/sentry/static/sentry/app/views/organizationDashboard/overviewDashboard.jsx [hunk 3]
```diff
@@ -20,7 +20,14 @@ class OverviewDashboard extends AsyncView {
   }
 
   renderBody() {
-    return <Dashboard releases={this.state.releases} {...overviewDashboard} />;
+    return (
+      <Dashboard
+        releases={this.state.releases}
+        busy={this.state.loading}
+        {...overviewDashboard}
+        {...this.props}
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/organizationDashboard/overviewDashboard.jsx]: Is it just router that we need from here now? If that's the case can we just pass `router={this.props.router}`. It's often difficult for the reader to figure out what's actually needed with {...this.props}

File: src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx]: Just curious, is there a specific reason why we are increasing from 10 to 20?

File: src/sentry/static/sentry/app/views/organizationDashboard/overviewDashboard.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDashboard/overviewDashboard.jsx]: Hmm it's needed to override `overviewDashboard` for tests as well :/ I've destructured it explicitly and added a comment: https://github.com/getsentry/sentry/pull/12861/commits/5fd64f8ec598baac219e5c83ae5724aef8e1dea6

File: src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx]: Was testing this for staging, will return it back to 10, but the idea is that if you release often and you're looking at a sufficiently long time period, your chart ends up looking like: 
image

File: src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationDashboard/discoverQuery.jsx]: I think the querying logic should be held in this component, its parents don't need to know details about the queries itself.

We could change the name of the `busy` prop to something more specific to releases state if that would help clarify things.

**Review by lynnagara — APPROVED** [review]

**Pull request merged**